### ISSUE TITLE
Support array-access syntax

### ISSUE THREAD

**Issue by cdmistman** [sagiegurari/cargo-make#402]
**Features Description**
There should be some kind of syntax for accessing arrays. For example, this would be useful following `@@split(string,delimiter)`.

**Describe the solution you'd like**
A simple square-bracket based array-access syntax following cargo make functions.

**Code Sample**
```toml
[env]
# This could be the name of a rustc-supported triplet or a custom-defined triplet
TARGET = { value = "default.json", condition = { env_not_set = "THING" } }

[tasks.print]
command = "echo"
args = ["now starting build for", "@@split(TARGET,.)[0]"]

[tasks.build]
dependencies = ["print"]
command = "cargo"
# TARGET here is used regardless of whether or not it is a file
args = ["build", "--target", "${THING}"]
```

**Comment by sagiegurari** [comment]
nice. i can create a new function that splits and gets like
getat(envvar,delimiter,index)
if its out of bounds the entire arg is removed (like split does if the env var is empty or doesn't exist)

**Comment by sagiegurari** [comment]
by the way, you can do something already today.
you could have some task that is invoking duckscript script that splits and takes the argument.
then it can set it as a new env var to be accessed by other tasks.

I will implement the new function, but just wanted to point out that you are not completely blocked at the moment.

**Comment by sagiegurari** [comment]
@cdmistman you can try the 'getat' function now available in the dev branch 0.30.2
it works like the split, but accepts a third param which is the index to return.

you can look at docs + example:

https://github.com/sagiegurari/cargo-make/tree/0.30.2#usage-functions-getat

**Comment by cdmistman** [comment]
thanks!

**Comment by sagiegurari** [comment]
this is now released in 0.30.2 cargo-make version.
thanks for the feedback.
